components/sections/settingsSection/idpSection/idpSectionComponent.tsx

```diff
@@ -0,0 +1,208 @@
+    useState<IdentityProviderTemplate>();
+
+  const templates: IdentityProviderTemplate[] = [
+    EnterpriseIdentityProvider,
+    GoogleIdentityProvider,
+  ];
+
+  const fetchAllIdPs = useCallback(async () => {
+    setLoadingDisplay(LOADING_DISPLAY_BLOCK);
+    const res = await listAllIdentityProviders(session!);
+
+    if (res) {
+      setIdpList(res);
+    } else {
+      setIdpList([]);
+    }
+    setLoadingDisplay(LOADING_DISPLAY_NONE);
+  }, [session]);
+
+  async function listAllIdentityProviders(
+    session: Session
+  ): Promise<IdentityProvider[] | null> {
+    try {
+      const body = {
+        orgId: session ? session.orgId : null,
+        session: session,
+      };
+
+      const request = {
+        body: JSON.stringify(body),
+        method: RequestMethod.POST,
+      };
+      const res = await fetch(
+        "/api/settings/identityProvider/listAllIdentityProviders",
+        request
+      );
+      const data = await res.json();
+
+      if (data) {
+        if (data.identityProviders) {
+          return data.identityProviders;
+        } else {
+          return [];
+        }
+      }
+      return data;
+    } catch (err) {
+      return null;
+    }
+  }
+
+  useEffect(() => {
+    fetchAllIdPs();
+  }, [fetchAllIdPs]);
+
+  const onAddIdentityProviderClick = (): void => {
+    setOpenSelectModal(true);
+  };
+
+  const onTemplateSelect = (template: IdentityProviderTemplate): void => {
+    setOpenSelectModal(false);
+    setSelectedTemplate(template);
+  };
+
+  const onSelectIdpModalClose = (): void => {
+    setOpenSelectModal(false);
+  };
+
+  const onCreationDismiss = (): void => {
+    setSelectedTemplate(undefined);
+  };
+
+  const onIdpCreated = (response: IdentityProvider | null): void => {
+    if (response) {
+      successTypeDialog(toaster, "Success", "Connection Created Successfully");
+
+      setIdpList([...idpList, response]);
+
+      setOpenSelectModal(false);
+      setSelectedTemplate(undefined);
+    } else {
+      errorTypeDialog(
+        toaster,
+        "Error Occured",
+        "Error occured while creating the connection. Try again."
+      );
+    }
+  };
+
+  return (
+    <div className={"mainPanelDiv"}>
+      <Container>
+        <Stack direction="row" justifyContent="space-between">
+          <Stack direction="column" alignItems="flex-start">
+            <h3>Login Connections</h3>
+            <p>
+              Manage Login Connections to allow users to log in to your
+              application through them.
+            </p>
+          </Stack>
+
+          {idpList && idpList.length !== 0 ? (
+            <Button
+              style={{ borderRadius: "50px" }}
+              appearance="primary"
+              size="md"
+              onClick={onAddIdentityProviderClick}
+            >
+              Add Connection
+            </Button>
+          ) : null}
+        </Stack>
+
+        {idpList ? (
+          idpList.length === 0 ? (
```

Comment:
we can use the optional chaining operator to avoid nested ternary expressions.

```suggestion
        {
          idpList?.length === 0 ? (
```

Reply:
Fixed